src/cachemachine/rubinimageinfo.py

```diff
@@ -0,0 +1,346 @@
+import re
+from dataclasses import dataclass
+from enum import Enum, auto
+from typing import Optional
+
+import semver
+from docker_image import reference
+
+from .types import IncomparableImagesError
+
+DEFAULT_TAG = "latest"  # Docker convention
+DEFAULT_RECOMMENDED_TAG = "recommended"
```

Comment:
Please don't have a variable that says recommended.  I'm passing what tag is the recommended in so that we don't have these kinds of defines and logic based on recommended